### PR TITLE
fix(userspace/engine): make sure exception fields are not optional in replace mode

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind bug

**Any specific area of the project related to this PR?**

/area engine

**What this PR does / why we need it**:

When replaced with the new `override` YAML construct, `exceptions` must not have any of their sub-fields as optional, just like when first defined in regular rules definitions.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix(userspace/engine): make sure exception fields are not optional in replace mode
```

### CODE REVIEW

**Comment by LucaGuerra** [comment]
Could you please add a test for this behavior? This way we won't break it in the future.

File: unit_tests/engine/test_rule_loader.cpp [hunk 1]
```diff
@@ -680,6 +680,42 @@ TEST_F(test_falco_engine, rule_override_with_enabled)
...
+  exceptions:
+    - name: test_exception
+      values: [id]
+  override:
+    exceptions: replace
+)END";
+
+	ASSERT_FALSE(load_rules(rules_content, "rules.yaml"));
+  ASSERT_FALSE(has_warnings());
+	ASSERT_TRUE(check_error_message("Item has no mapping for key 'fields'")) << m_load_result_json.dump();
```
Comment (LucaGuerra) [inline unit_tests/engine/test_rule_loader.cpp:716]: lint? otherwise looks good, thanks!

**Review by LucaGuerra — APPROVED** [review]

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**